### DISCUSSION TITLE
Zitadel OIDC discovery url not accessible when configured in route of APISIX gateway when both apps are in same cluster

### DISCUSSION THREAD

**Discussion by thirumurthis** [zitadel/zitadel#7938] [Q&A]
Deployed Zitadel and Apisix in kind cluster, and was able to access the UI's from the cluster.

The Zitadel is deployed to the cluster with `ExternalDomain` as `zitadel.localhost`. The Apisix route configuration, OIDC plugin was configured with the discovery url `http://zitadel.localhost`, but Apisix pod is not able to access it within the kubernetes cluster.

- When I add the `http://<zitadel service>.<zitadel namespace>` in Apisix plugin notice the Zitadel discovery url is accessible, but Zitadel is not able to process since the domain name didn't match with ExternalDomain.

- Not sure if there is a way to add list of `ExternalDomain`  which includes `zitadel.localhost` and `<zitadel service>.<zitadel namespace>` so zitadel can be accessible.

refer [Stackoverflow](https://stackoverflow.com/questions/78463238/apisix-route-oidc-plugin-to-connect-to-zitadel-odic-not-accessible-within-the-sa)

**Comment by fforootd** [comment]
Can you share the full error you see in apisix?

To me it looks like apisix does not use the host header in the call to zitadel correctly.

**Reply by thirumurthis** [reply]
I was using the default configuration for Apisix discovery. For testing with kind cluster, I set the discovery registry to Kubernetes and with the helm release name being zitadel and namespace set to localhost for zitadel helm deployment. Was able toale it work for the domain `zitadel.localhost`. Apisix configuration for discovery has registry for DNS which I need to validate.
